app/components/activity/index.ts

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+import template from './template.html!text';
+
+export default function(app) {
+  app.component('activity', {
+    bindings: {
+      discussion: '<',
+      document: '<',
+      user: '<',
+    },
+    controller: [
+    '$http', 'config', 'notificationService',
+      function($http, config, notificationService) {
+        const ctrl = this;
+        ctrl.$onChanges = function(changesObj) {
```

Comment:
I suggest to use arrow functions for everything (except the controller function itself).